Minor changes

jonasfreimuth opened the pull request:
Minor fixes and changes. Will get the pipeline to the state described in PR BIMSBbioinfo/pigx_docs#15 of the pigx_docs.

jonasfreimuth committed:
add creation of fastp output dir variable

jonasfreimuth committed:
change fastp report output to go into own dir

  This should reduce confusion when looking for fastp results.

jonasfreimuth committed:
fix genome coverage file naming

  Is actually a tab delimited file.

jonasfreimuth committed:
fix mutation coverage file naming.

  Is actually a tab delimted file.

jonasfreimuth committed:
Reflow markdown in qc_report_per_sample.Rmd

jonasfreimuth committed:
Remove unnecessary saving of sample quality df

  Also remove all the infrastructure relating to the output path of
  this file. The additional saving has not been necessary for a while I
  think.

jonasfreimuth committed:
rename variantreport script to be consistent

  Admittedly pretty nitpicky, but I want to tidy up before the next
  release.

jonasfreimuth committed:
add built tarball to .gitignore

jonasfreimuth committed:
add python build dirs to .gitignore

  Sparked by `__pycache__`, genrated by `download_tarball.py`.

jonasfreimuth committed:
correct mutation count file name

jonasfreimuth committed:
rename mut count col for sign incr mutations

  Use was not clear from the name.

jonasfreimuth committed:
fix locations in overall settings.yaml

  Now the tests/sample_data locations are hardcoded there. I think it
  should be discussed whether these locations should just be removed
  from the overall sample sheet to only be provided in the test specific
  sample sheets, so that it would result in an error if they are not
  expliticly provided by the user. That should prevent a situation where
  the wrong input file is quietly used. However I also think that should
  be implemented only together with explicit error handling.

jonasfreimuth committed:
Add .in settings file tesing pipeline setup

  As opposed to testing integrity of the code, this is supposed to test
  everything works after an installation.

jonasfreimuth committed:
Add preconfigured setup testing settings file

  For the case that the pipeline is being used prebuilt and the tests
  dir is being used as a standalone.

jonasfreimuth committed:
Add setup_test_settings.yaml.in to Makefile.am

jonasfreimuth committed:
Add setup_test_settings.yaml.in to configure.ac

jonasfreimuth committed:
change .gitignore to ignore output* dirs in tests/

  Done to include the setup-test outpu dir.

jonasfreimuth committed:
move deconvolution settings into parameters

jonasfreimuth committed:
Extend .gitignore entry to include distcheck dir

jonasfreimuth committed:
move deconvolution: method comment to right place

jonasfreimuth committed:
Move mutation_depth_threshold to deconv params

  That is the more intuitive place for it I think. The parameter
  affects actual computation not just reported results.

jonasfreimuth committed:
Rename primer location bed file

  New name should be way clearer.

jonasfreimuth committed:
Fix check for empty variables in vcfTocsv.py

jonasfreimuth committed:
Remove unused snakefile script vep_rule.py

jonasfreimuth committed:
Rename lofreq output vcf files

  Done so it is easier to use external vcf files.

jonasfreimuth committed:
Simplify vcf2csv_input function

jonasfreimuth committed:
Simplify vep_input function

jonasfreimuth committed:
Move cluster sub setting to other cluster settings

  Just for nicer layout.

jonasfreimuth committed:
Remove unused nice setting from etc/settings.yaml

rekado commented on tests/setup_test_settings.yaml:
```diff
@@ -0,0 +1,18 @@
+locations:
```
This file should not be included.  It should be generated from the corresponding .in file.

jonasfreimuth replied:
It is explicitly included to allow simple downloading of the repo and running a setup check with manually installed databases *without* entering of a build environment and configuring. I have clarified that [section](https://jonasfreimuth.github.io/pigx_docs/pigx-sars-cov-2.html#quick-start) in the docs a bit.

rekado replied:
Okay, got it.

rekado commented on snakefile.py:
```diff
@@ -376,6 +374,7 @@ def render_qc_report_params(wildcards, input, output = None, threads = None, res
 COVERAGE_DIR      = os.path.join(OUTPUT_DIR, 'coverage')
 REPORT_DIR        = os.path.join(OUTPUT_DIR, 'report')
+FASTP_DIR         = os.path.join(OUTPUT_DIR, 'fastp')
 FASTQC_DIR        = os.path.join(REPORT_DIR, 'fastqc')
```
Is FASTQC_DIR still used anywhere?  Or is it all FASTP_DIR now?

jonasfreimuth replied:
Yes, I just wanted to make sure that the fastp output can be easily found. I felt it was confusing to put the output of one tool into the dir of another.

rekado commented on snakefile.py:
```diff
@@ -321,7 +319,7 @@ def render_qc_report_params(wildcards, input, output = None, threads = None, res
 INPUT_DIR           = config["locations"]["input-dir"]
 REFERENCE_FASTA     = config["locations"]["reference-fasta"]
-AMPLICONS_BED       = config["locations"]["amplicons-bed"]
+PRIMERS_BED       = config["locations"]["primers-bed"]
```
nitpick: indentation

jonasfreimuth replied:
Fixed

rekado commented on etc/settings.yaml.in:
```diff
@@ -50,29 +51,29 @@ parameters:
+    #   qp:
+    #     Quadratic programming (`deconvR`).
+    #   svr:
+    #     Support vector regression (`deconvR`).
```
I'm not a fan of moving the comments inside the block, but I don't feel strongly enough about this to argue :)

jonasfreimuth replied:
Hmm I removed all the comments, as everything should be explained in the docs anyways...

rekado reviewed (changes requested):
The generated yaml file should be excluded.  Otherwise this looks good to me.

jonasfreimuth committed:
Fix indentation for primers bed config reading

jonasfreimuth committed:
Remove comments in settings.yaml & point to docs

  Users shouldn't have to go into program files for information, and
  developers shouldn't have two places to correct when things are
  changed.
